### PR TITLE
[GLUTEN-5414] [VL] Support arrow csv option and schema

### PR DESCRIPTION
**Pull request by jinchengchenghh**
Support basic option now, will support more options after arrow patch merged.
https://github.com/apache/arrow/pull/41646
Before this patch, if the required schema is different with file schema, csv read will fallback.
And changed to use index in file instead of check the file column name considering case sensitive.
Add a new common test function when the rule applies to Logical plan.

Compile arrow with version 15.0.0-gluten, upgrade arrow-dataset and arrow-c-data version from 15.0.0 to 15.0.0-gluten

### CODE REVIEW

**Comment by jinchengchenghh** [comment]
Can you help review this PR and rerun the failed test? Thanks! @zhztheplayer

**Review by zhztheplayer — COMMENTED** [review]
Just some nits. Thanks!

File: backends-velox/src/test/resources/datasource/csv/student_option.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Name;Language
+Juno;Java
+Peter;Python
+Celin;C++
```
[the remark this answers is not in the document]
Comment (jinchengchenghh) [reply backends-velox/src/test/resources/datasource/csv/student_option.csv:4]: The csv file is not relevant to arrow datasource, it is just used to test Arrow.
It is just a normal CSV file, so I place it here.

**Review by zhztheplayer — APPROVED** [review]

**Comment by jinchengchenghh** [comment]
Can you help merge this one? Thanks!

**Review by zhztheplayer — APPROVED** [review]
Thanks!

File: ep/build-velox/src/build_velox.sh [hunk 2]
```diff
@@ -282,9 +282,11 @@ function compile_arrow_java_module() {
...
 
     pushd $ARROW_HOME/java
-    
-    mvn clean install -pl maven/module-info-compiler-maven-plugin -am \
-          -Dmaven.test.skip -Drat.skip -Dmaven.gitcommitid.skip -Dcheckstyle.skip
+    # Because arrow-bom module need the -DprocessAllModules
+    mvn versions:set -DnewVersion=15.0.0-gluten -DprocessAllModules
+   
+    mvn clean install -am \
+          -DskipTests -Drat.skip -Dmaven.gitcommitid.skip -Dcheckstyle.skip
```
Comment (Yohahaha) [inline ep/build-velox/src/build_velox.sh:289]: do we really need compile whole project? it's a time consuming command...
I wonder which depends module can not be compiled by `-am` options?

Comment (jinchengchenghh) [reply ep/build-velox/src/build_velox.sh:289]: arrow-bom module includes many modules, so we need to compile whole project

**Pull request merged**